src/test/scala-2/chiselTests/ContractSpec.scala

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.ltl._
+import chisel3.testers.BasicTester
+import chisel3.experimental.SourceLine
+import _root_.circt.stage.ChiselStage
+import chiselTests.ChiselRunners
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import Sequence._
+
+class ContractSpec extends AnyFlatSpec with Matchers with FileCheck {
+  it should "support contracts with no operands" in {
+    class Foo extends RawModule {
+      val a = IO(Input(Bool()))
+      Contract() {
+        AssertProperty(a)
+      }
+    }
+    generateFirrtlAndFileCheck(new Foo)("""
+      // CHECK-LABEL: module Foo
+      // CHECK: input a : UInt<1>
+      // CHECK: contract :
+      // CHECK:   intrinsic(circt_verif_assert, a)
+      """)
+  }
+
+  it should "support contracts with a single operand" in {
+    class Foo extends RawModule {
+      val a = IO(Input(Bool()))
+      val b = Contract(a) { x =>
```

Comment:
i am missing the basic user understanding of when something should go into the argument list of the `Contract` or just be captured in the closure of the body.